gen/suites/actor_creation/params.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+
+	. "github.com/filecoin-project/test-vectors/gen/builders"
+)
+
+func createActorUnparsableParams(v *Builder) {
+	v.Messages.SetDefaults(GasLimit(1_000_000_000), GasPremium(1), GasFeeCap(200))
+
+	sender := v.Actors.Account(address.SECP256K1, abi.NewTokenAmount(1_000_000_000_000))
+	receiver := v.Actors.Account(address.SECP256K1, abi.NewTokenAmount(1_000_000_000_000))
+
+	v.CommitPreconditions()
+
+	balanceBefore := v.Actors.Balance(sender.Robust)
+
+	// Valid message for construction of a payment channel
+	createMsg := v.Messages.Sugar().CreatePaychActor(sender.Robust, receiver.Robust, Value(abi.NewTokenAmount(10_000)), Nonce(0))
+
+	// Form an invalid CBOR payload
+	createMsg.Message.Params = append([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, createMsg.Message.Params...)
+
+	v.CommitApplies()
+
+	v.Assert.EveryMessageResultSatisfies(ExitCode(exitcode.SysErrSenderInvalid))             // make sure that we get SysErrSenderInvalid error code
```

Comment:
This exit code is still wrong, should be `ErrSerialization`.

Of particular note, if it did return `SysErrSenderInvalid` that would imply that the gas should not **not** be deducted from the sender's account below, so this is doubly wrong.

Reply:
So there are two cases we might want to test:
1. parameters to InitExec are garbled.
2. parameters to constructor are garbled.

This is testing (1), and I can confirm that the VM fails due to a failure to deserialize:

![image](https://user-images.githubusercontent.com/1101242/91050491-a29ba600-e616-11ea-9cb8-c2b339771cef.png)

Note the retCode argument to `Absorb` is indeed 1 (`SysErrSenderInvalid`).

I can see (2) being `ErrSerialization` (actor-level error code), but should this be `SysErrSerialization`? (It's a VM level system error).

Reply:
Confirmed that this is returning `SysErrSenderInvalid` and deducting the gas costs.

Reply:
I've added tests for both cases (arguments to `InitExec` corrupted; arguments to constructor corrputed); both marked as `incorrect`.

Reply:
Reported to Lotus: https://github.com/filecoin-project/lotus/issues/3264.

Reply:
`SysErrSerialization` should be removed, it's _always_ `ErrSerialization`.

Logically, the VM interface is just bytes, and serialization is actor code. For convenience, we do the CBOR inside the VM boundary. But the confusion this causes is beginning to outweigh the convenience.